src/EventSubscriber/WebSSOEventSubscriber.php

```diff
@@ -0,0 +1,312 @@
+     * @param AuthenticationRepositoryInterface $authenticationRepository
+     * @param Security $security
+     */
+    public function __construct(
+        private int $sessionExpirationDelay,
+        private Container $dependencyInjector,
+        private ReadWebSSOConfigurationRepositoryInterface $webSSOReadRepository,
+        private ContactRepositoryInterface $contactRepository,
+        private SessionInterface $session,
+        private AuthenticationServiceInterface $authenticationService,
+        private SessionRepositoryInterface $sessionRepository,
+        private DataStorageEngineInterface $dataStorageEngine,
+        private OptionServiceInterface $optionService,
+        private AuthenticationRepositoryInterface $authenticationRepository,
+        private Security $security,
+    ) {
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => [
+                ['loginWebSSOUser', 34]
+            ],
+        ];
+    }
+
+    /**
+     * login User with Web SSO
+     *
+     * @param RequestEvent $event
+     */
+    public function loginWebSSOUser(RequestEvent $event): void
+    {
+        if ($this->security->getUser() === null) {
+            $request = $event->getRequest();
+            $webSSOConfiguration = $this->findWebSSOConfigurationOrFail();
+            if ($webSSOConfiguration->isActive()) {
+                $this->validateIpIsAllowToConnect($request->getClientIp(), $webSSOConfiguration);
+                $this->validateLoginAttributeOrFail($webSSOConfiguration);
+
+                $userAlias = $_SERVER[$webSSOConfiguration->getLoginHeaderAttribute()];
+                if ($webSSOConfiguration->getPatternMatchingLogin() !== null) {
+                    $userAlias = $this->extractUsernameFromLoginClaimOrFail($webSSOConfiguration);
+                }
+                $user = $this->findUserByAliasOrFail($userAlias);
+                $this->createSession($user, $request);
+                $sessionId = $this->session->getId();
+                $request->headers->set('Set-Cookie', "PHPSESSID=" . $sessionId);
+                $this->createTokenIfNotExist($sessionId, $webSSOConfiguration->getId(), $user, $request->getClientIp());
+            }
+        }
+    }
+
+    /**
+     * Extract username using configured regexp for login matching
+     *
+     * @param WebSSOConfiguration $webSSOConfiguration
+     * @return string
+     * @throws SSOAuthenticationException
+     */
+    private function extractUsernameFromLoginClaimOrFail(WebSSOConfiguration $webSSOConfiguration): string
+    {
+        $userAlias = preg_replace(
+            '/' . trim($webSSOConfiguration->getPatternMatchingLogin(), '/') . '/',
+            $webSSOConfiguration->getPatternReplaceLogin() ?? '',
+            $_SERVER[$webSSOConfiguration->getLoginHeaderAttribute()]
+        );
+        if (is_array($userAlias) || empty($userAlias)) {
```

Comment:
since first argument of preg_replace is a string, `$userAlias` can't be an array
Is it a case which phpstan does not handle ?

Reply:
hmm nope, misunderstanding of me, didn't notice this in PHP documentation 
`preg_replace() returns an array if the subject parameter is an array, or a string otherwise.`